src/common/utils/formatter/formatter.ts

```diff
@@ -139,22 +140,78 @@ export function getFormattedClause(dimension: Dimension, clause: FilterClause, t
       if (clause.action === Filter.CONTAINS) values = `"${values}"`;
 
       break;
-
     case 'time':
-      var selection = (clause.selection as LiteralExpression);
-      var selectionType = selection.type;
-      if (selectionType === 'TIME_RANGE') {
-        var timeRange = selection.value as TimeRange;
-        values = formatTimeRange(timeRange, timezone, DisplayYear.IF_DIFF);
-      } else if (selectionType === "SET/TIME") {
-        values = clauseSet.toString();
-      }
+      values = getFormattedTimeClause(clause, timezone);
       break;
-
     default:
       throw new Error(`unknown kind ${dimKind}`);
   }
 
-  return {title: getClauseLabel(), values};
+  return { title: getClauseLabel(), values };
+}
+
+export function getFormattedTimeClause(clause: FilterClause, timezone: Timezone): string {
```

Comment:
is it feasible to move this method into FilterClause class?

Reply:
this method is specific to time dimension clauses, to make it good it would IMHO require having specific FilterClauses by dimension type